rclpy/rclpy/parameter_event_handler.py

```diff
@@ -0,0 +1,367 @@
+                        event, parameter_name=param_name, node_name=node_name
+                    ):
+                        for callback_handle in callbacks_list:
+                            with callback_handle.mutex:
+                                callback_handle.callback(parameter)
+
+                for event_callback in self.event_callbacks:
+                    with event_callback.mutex:
+                        event_callback.callback(event)
+
+        def add_parameter_callback(
+            self,
+            parameter_name: str,
+            node_name: str,
+            callback: Callable[[Parameter], None],
+        ) -> ParameterCallbackHandle:
+            """
+            Add new parameter callback.
+
+            Callbacks are called in FILO manner.
+
+            :param parameter_name: Name of a parameter to bind the callback to
+            :param node_name: Name of a node, that the parameter should be related to
+            :param callback: A callable to be referenced when parameter event occurs
+
+            :return ParameterCallbackHandle: The duplicate of a parameter callback handle,
+                stored within Callbacks object
+            """
+            handle = ParameterCallbackHandle(
+                parameter_name=parameter_name,
+                node_name=node_name,
+                callback=callback,
+            )
+
+            with self.mutex:
+                self.parameter_callbacks[(parameter_name, node_name)].insert(0, handle)
+
+            return handle
+
+        def remove_parameter_callback(
+            self,
+            handle: ParameterCallbackHandle,
+        ) -> None:
+            """
+            Remove the parameter callback related to provided handle.
+
+            :param handle: A handle of the callback that is to be removed
+            """
+            with self.mutex:
+                handle_key = (handle.parameter_name, handle.node_name)
+                if handle_key in self.parameter_callbacks:
+                    if handle in self.parameter_callbacks[handle_key]:
+                        self.parameter_callbacks[handle_key].remove(handle)
+                    else:
+                        raise RuntimeError("Callback doesn't exist")
+
+                    if len(self.parameter_callbacks[handle_key]) == 0:
+                        self.parameter_callbacks.pop(handle_key)
+                else:
+                    raise RuntimeError("Callback doesn't exist")
+
+        def add_parameter_event_callback(
+            self,
+            callback: Callable[[ParameterEvent], None],
+        ) -> ParameterEventCallbackHandle:
+            """
+            Add new parameter event callback.
+
+            Callbacks are called in FILO manner.
+
+            :param callback: A callable to be referenced on a ParameterEvent
+
+            :return ParameterEventCallbackHandle: The duplicate of a callback handle,
+                stored in a parameter_event dict
```

Comment:
```suggestion
        :return ParameterEventCallbackHandle: A handle that should be saved by the user so that the event_callback can be later removed.
```